src/class_sqlThread.py

```diff
@@ -26,117 +26,145 @@
+            os._exit(0)
         except Exception as err:
-            if str(err) == 'table inbox already exists':
-                logger.debug('Database file already exists.')
-
-            else:
-                sys.stderr.write(
-                    'ERROR trying to create database file (message.dat). Error message: %s\n' % str(err))
-                os._exit(0)
-
-        # If the settings version is equal to 2 or 3 then the
-        # sqlThread will modify the pubkeys table and change
-        # the settings version to 4.
-        settingsversion = BMConfigParser().getint(
-            'bitmessagesettings', 'settingsversion')
-
-        # People running earlier versions of PyBitmessage do not have the
-        # usedpersonally field in their pubkeys table. Let's add it.
-        if settingsversion == 2:
-            item = '''ALTER TABLE pubkeys ADD usedpersonally text DEFAULT 'no' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 3
-
-        # People running earlier versions of PyBitmessage do not have the
-        # encodingtype field in their inbox and sent tables or the read field
-        # in the inbox table. Let's add them.
-        if settingsversion == 3:
-            item = '''ALTER TABLE inbox ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE inbox ADD read bool DEFAULT '1' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE sent ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 4
+            sys.stderr.write(
+                'ERROR trying to upgrade database. Error message: %s\n' % str(err))
+            os._exit(0)
+
+    @property
+    def sql_schema_version(self):
+        """
+            Getter for get current schema version
+        """
+        return self.__get_current_settings_version()
+
+    @sql_schema_version.setter
+    def sql_schema_version(self):
+        """
+            Update version with one level
+        """
+        item = '''update settings set value=? WHERE key='version';'''
+        self.cur.execute(item, self.__get_current_settings_version() + 1)
+
+    def upgrade_to_latest(self):
+        """
+            Initialise upgrade level
+        """
+
+        # Declare variables
+        # self.conn = conn
+        # self.cur = cur
+        # if current_level:
+        #     self._current_level = current_level
+        # else:
+        #     self._current_level = self.__get_current_settings_version()
+        # if max_level:
+        #     self.max_level = max_level
+
+        while self.__get_current_settings_version() < self.max_level:
```

Comment:
use getter

Reply:
```
increment = True

while self.settings_version < self.max_level:
   self._upgrade()
   self.settings_version = increment
```

and create an empty sql script for level 1 (one that doesn't change anything), like `SELECT NULL`